Microsoft.Toolkit.Uwp.UI.Controls/Carousel/CarouselItemAutomationPeer.cs

```diff
@@ -0,0 +1,147 @@
+                return peer != null ? this.ProviderFromPeer(peer) : null;
+            }
+        }
+
+        private CarouselItem OwnerCarouselItem
+        {
+            get { return this.Owner as CarouselItem; }
+        }
+
+        /// <summary>Adds the current element to the collection of selected items.</summary>
+        public void AddToSelection()
+        {
+            CarouselItem owner = this.OwnerCarouselItem;
+            Carousel parent = owner.ParentCarousel;
+            parent?.SetSelectedItem(owner);
+        }
+
+        /// <summary>Removes the current element from the collection of selected items.</summary>
+        public void RemoveFromSelection()
+        {
+            CarouselItem owner = this.OwnerCarouselItem;
+            Carousel parent = owner.ParentCarousel;
+            if (parent != null)
+            {
+                parent.SelectedItem = null;
+            }
+        }
+
+        /// <summary>Clears any existing selection and then selects the current element.</summary>
+        public void Select()
+        {
+            CarouselItem owner = this.OwnerCarouselItem;
+            Carousel parent = owner.ParentCarousel;
+            parent?.SetSelectedItem(owner);
+        }
+
+        /// <summary>
+        /// Gets the control type for the element that is associated with the UI Automation peer.
+        /// </summary>
+        /// <returns>The control type.</returns>
+        protected override AutomationControlType GetAutomationControlTypeCore()
+        {
+            return AutomationControlType.ListItem;
+        }
+
+        /// <summary>
+        /// Called by GetClassName that gets a human readable name that, in addition to AutomationControlType,
+        /// differentiates the control represented by this AutomationPeer.
+        /// </summary>
+        /// <returns>The string that contains the name.</returns>
+        protected override string GetClassNameCore()
+        {
+            return Owner.GetType().Name;
+        }
+
+        /// <summary>
+        /// Called by GetName.
+        /// </summary>
+        /// <returns>
+        /// Returns the first of these that is not null or empty:
+        /// - Value returned by the base implementation
+        /// - Name of the owning CarouselItem
+        /// - Carousel class name
+        /// </returns>
+        protected override string GetNameCore()
+        {
+            int? index = this.OwnerCarouselItem.ParentCarousel?.IndexFromContainer(this.OwnerCarouselItem);
+
+            string name = base.GetNameCore();
+            if (!string.IsNullOrEmpty(name))
+            {
+                return $"{name}";
+            }
+
+            if (this.OwnerCarouselItem != null && !string.IsNullOrEmpty(this.OwnerCarouselItem.Name))
+            {
+                return this.OwnerCarouselItem.Name;
+            }
+
+            if (string.IsNullOrEmpty(name))
+            {
+                name = this.GetClassName();
```

Comment:
Same here, narrator should already announce the class/type. Announcing it twice doesn't help users much.

Reply:
Updated this also to return the name of the control if provided, otherwise it will look for a `TextBlock` if there is one in the item's template and return that text, otherwise will fallback to the base implementation.